src/main/java/com/microsoft/azure/functions/worker/handler/FunctionEnvironmentReloadRequestHandler.java

```diff
@@ -43,36 +50,49 @@ FunctionMethodDescriptor createFunctionDescriptor(String functionId,
    * if java language worker is started in standby mode by the functions runtime
    * and should not be used for other purposes
    */
-  public static void setEnv(Map<String, String> newSettings) throws Exception {
+  public void setEnv(Map<String, String> newSettings) throws Exception {
     if (newSettings == null || newSettings.isEmpty()) {    
       return;
     }
-	  try {
-	    Class<?> processEnvironmentClass = Class.forName("java.lang.ProcessEnvironment");
-	    Field theEnvironmentField = processEnvironmentClass.getDeclaredField("theEnvironment");
-	    theEnvironmentField.setAccessible(true);
-	    Map<String, String> env = (Map<String, String>) theEnvironmentField.get(null);
-	    env.putAll(newSettings);
-	    Field theCaseInsensitiveEnvironmentField = processEnvironmentClass.getDeclaredField("theCaseInsensitiveEnvironment");
-	    theCaseInsensitiveEnvironmentField.setAccessible(true);
-	    Map<String, String> cienv = (Map<String, String>)     theCaseInsensitiveEnvironmentField.get(null);
-	    cienv.putAll(newSettings);
-	  } catch (NoSuchFieldException e) {
-	    Class[] classes = Collections.class.getDeclaredClasses();
-	    Map<String, String> env = System.getenv();
-	    for(Class cl : classes) {
-	      if("java.util.Collections$UnmodifiableMap".equals(cl.getName())) {
-	        Field field = cl.getDeclaredField("m");
-	        field.setAccessible(true);
-	        Object obj = field.get(env);
-	        Map<String, String> map = (Map<String, String>) obj;
-	        map.clear();
-	        map.putAll(newSettings);
-	      }
-	    }
-	  }
-	}
 
+    for (Map.Entry<String,String> entry : newSettings.entrySet()) {
+        Kernel32.INSTANCE.SetEnvironmentVariable(entry.getKey(), entry.getValue());
+      }
+
+      Kernel32.INSTANCE.SetEnvironmentVariable("WEBSITE_PLACEHOLDER_MODE", null);
```

Comment:
two calls required? Can't we just set `WEBSITE_PLACEHOLDER_MODE` to 0 directly?

Reply:
Basically we will need to do a merge here. meaning delete the env variables which don't exist anymore by setting them to null. update the existing ones and add the new ones. so this part needs to be updated.